code/common/os.q

```diff
@@ -7,10 +7,12 @@ ext:{`$(x?".")_(x:string x;x)[i:10h=type x]}
 del:{system("rm ";"del ")[NT],pth x}
 deldir:{system("rm -r ";"rd /s /q ")[NT],pth x}
 hdeldir:{[dirpath]
+ dirpath:@[{[x] $[10h~a:type x;x;-11h~a;string x;'`err]};dirpath;{.lg.e[`hdeldir;"Invalid data type for directory path"]}];
+ if[()~dirpath;:()];
  diR:{$[11h=type d:key x;raze x,.z.s each` sv/:x,/:d;d]};
- nuke:hdel each desc diR@;
- nuke hsym`$dirpath;
- .lg.o[`deldir;"deleting from  directory"]}
+ nuke:hdel each desc 1_diR@;
```

Comment:
Is there a case where we could want to also delete the parent dir? Not in finspace but in std torq maybe?
Perhaps this should be a bool input to the function.
`deleteparentdir` or something. Would be false for finspace. And we would only do the `1_` if the bool is false.